web/profiles/joinup/src/Access/CanonicalRouteRestrict.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup\Access;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\rdf_entity\RdfInterface;
+
+/**
+ * Restricts canonical routes to UID 1 only.
+ */
+class CanonicalRouteRestrict implements AccessInterface {
+
+  /**
+   * Constructs a CanonicalRouteRestrict object.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The route match object to be checked.
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The account being checked.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  public function access(RouteMatchInterface $route_match, AccountInterface $account): AccessResultInterface {
+    $rdf_entity = $route_match->getParameter('rdf_entity');
+    if (!($rdf_entity instanceof RdfInterface) || $rdf_entity->bundle() !== 'spdx_licence') {
+      // In case the route the canonical route of the spdx_licence, return a
+      // positive access result as a neutral would deny the access to the route.
+      // @see \Drupal\Core\Access\AccessResult::andIf().
+      return AccessResult::allowed()->addCacheContexts(['user']);
+    }
+
+    if ($account->id() === 1) {
+      return AccessResult::allowed()->addCacheContexts(['user']);
```

Comment:
Using the `user` cache context here is wrong because this would vary the cache too much, basically will create a cache version for each user hitting this route. Instead, `user.is_super_user` should have been used. But if we switch to the ăroposed permission check, the it can be `user. permissions`. See https://www.drupal.org/docs/8/api/cache-api/cache-contexts.

There's also another issue with the cache contexts: shouldn't we vary also on route name (`route` context)? We should test if a user denied a SPDX licence route will not be denied also on a different bundle page.